File: AndroidJSSpike/JSBridge/assets/domain/network_service.js
network = function(){
    var service = tw_networkService;
    var success = {};

    function ajax(url, successHandler) {
      network.success.successHandler1 = function( result ) {
        out.println("inside callback handler");
        successHandler( result );
      };
      out.println("inside networkservice");
      service.ajax(url, "network.success.successHandler1" );
    }

	return {
      ajax: ajax,
      success: success
	};
}();

File: AndroidJSSpike/JSBridge/src/com/thoughtworks/jsbridge/NetworkService.java
package com.thoughtworks.jsbridge;

import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.Log;

public class NetworkService {
    private static final String TAG = "NetworkService";
    private final Handler handler;

    public NetworkService(Handler handler) {
        this.handler = handler;
    }

    public void ajax(String url, String jsHandler) {
        Log.d(TAG, "Calling Android Network Service - " + url);
        Log.d(TAG, "With handler - " + jsHandler);

        Message msg = new Message();
        Bundle data = new Bundle();
        data.putString("url", url);
        data.putString("callback", jsHandler);
        msg.setData(data);
        handler.dispatchMessage(msg);
    }
}
